keyboards/keychron/c3_pro/c3_pro.c

```diff
@@ -0,0 +1,114 @@
+/* Copyright 2023 @ Keychron (https://www.keychron.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "c3_pro.h"
+
+static uint32_t os_switch_timer_buffer   = 0;
+static uint8_t  os_switch_indicate_count = 0;
+static uint8_t  led_toggle_count         = 5;
+
+void keyboard_post_init_kb(void) {
+    setPinOutputPushPull(LED_MAC_OS_PIN);
+    setPinOutputPushPull(LED_WIN_OS_PIN);
+    writePin(LED_MAC_OS_PIN, !LED_PIN_ON_STATE);
+    writePin(LED_WIN_OS_PIN, !LED_PIN_ON_STATE);
+
+    layer_state_t last_layer = eeconfig_read_default_layer();
+    if (last_layer) {
+        default_layer_set(last_layer);
+    } else {
+        default_layer_set(1U << 2);
+    }
+
+    keyboard_post_init_user();
+}
+
+void eeconfig_init_kb(void) {
+#if (EECONFIG_KB_DATA_SIZE) == 0
+    // Reset Keyboard EEPROM value to blank, rather than to a set value
+    eeconfig_update_kb(0);
+#endif
+    keymap_config.raw  = eeconfig_read_keymap();
+    keymap_config.nkro = 1;
```

Comment:
setting force_nkro would accomplish this, as well, without the additional code

Reply:
Hi, we want the keyboard to remain nkro state when Ex-factory or after resetting and still keep the  current state while plugging in the keyboard after switching between the 6nkro and nkro.